src/main/java/controllers/TrainerController.java

```diff
@@ -54,4 +57,36 @@ public TrainerData getTrainer(String userIdentifier) throws UserDoesNotExistErro
 
         return trainer;
     }
+
+    /**
+     * Searches for all users with the given first name.
+     * @param firstName The first name of the member.
+     * @return A list of all the members with the given first name.
+     */
+    public List<MemberInfo> searchForMember(String firstName) {
```

Comment:
Instead of only searching by first name, can we search by first name and last name? 

Examples:
```
searchForMember("Tom", "Black") -> List containing members with name Tom Black
searchForMember("Tom", "") -> List containing members with first name Tom
searchForMember("", "Black") -> List containing members with last name Black
```

Reply:
+1

Reply:
added search by first, last, both, neither (returns all members)